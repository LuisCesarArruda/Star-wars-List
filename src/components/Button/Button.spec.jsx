import userEvent from "@testing-library/user-event"
import  {Button} from "./index"

const { render, screen } = require("@testing-library/react")

describe("<Button/>", ()=>{

    it("should render the button with the text", ()=>{
        render(<Button text="load more"/>)

        const button = screen.getByRole("button",{name: /load more/i})

        expect(button).toBeInTheDocument()
    })

    it("Should call function on button click", () =>{
        const fn = jest.fn();

        render(<Button text= "load more" onClick={fn}/>)

        const button = screen.getByRole("button", {name: /load more/i})
        
        userEvent.click(button)

        expect(fn).toHaveBeenCalledTimes(1)

    })

    it("Should be disabled when disabled is true", () =>{

        render(<Button text= "load more" disabled= {true}/>)

        const button = screen.getByRole("button", {name: /load more/i} )
        expect(button).toBeDisabled()
    })

    it("Should match snapshot", () =>{

        const fn = jest.fn()

        const {container}=render(<Button text= "load more" onClick={fn} disabled= {true}/>)

        expect(container.childNodes).toMatchSnapshot()
    })
})