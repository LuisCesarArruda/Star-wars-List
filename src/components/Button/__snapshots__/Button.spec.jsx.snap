// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should match snapshot 1`] = `
NodeList [
  <button
    class="btn"
    disabled=""
  >
    load more
  </button>,
]
`;
